detox/src/android/expect.js

```diff
@@ -268,6 +277,12 @@ class Element {
     this._selectElementWithMatcher(this._originalMatcher._avoidProblematicReactNativeElements());
     return await new ActionInteraction(this._invocationManager, this, new SwipeAction(direction, speed, percentage)).execute();
   }
+
+  async captureBitmap() {
+    // TODO this should be moved to a lower-layer handler of this use-case
+    const resultBase64 = await new ActionInteraction(this._invocationManager, this, new CaptureBitmapAction()).execute();
+    return ezFile.saveRawBase64Data(resultBase64, { fileSuffix: '.detox.element.png' });
```

Comment:
Could not we just move `detox/src/artifacts/utils/temporaryPath.js` to `detox/src/utils`, import it:

```js
const temporaryPath = require('../utils/temporaryPath');
```

and just use in `captureBitmap`:
```js
const resultBase64 = await new ActionInteraction(this._invocationManager, this, new CaptureBitmapAction()).execute();
const pngPath = temporaryPath.for.png();
await fs.writeFile(pngPath, resultBase64, 'base64');
return pngPath;
```

and not create yet another mini-module, please?

Reply:
I considered doing that a number of times, but `temporaryPath` seemed very artifact oriented, seeing that the API set basically pairs 1:1 with the artifacts types we have (image, log, video, perf recording). If you find it apt nevertheless, I *will* use it here.

The mini-module was a compromise as we do not have a lower-layer logic-provider for this, just yet. I wanted to keep the logic here down to the bare minimum. 'Makes it easier to test (when in a separate module), too. wdyt?

Reply:
I just don't see what it is here to test. :man_shrugging:

The test idea here would be to check whether we write a received (mocked) base64 string to some temporary file.

If we extract logic to a separate file, it feels like we are going to write very similar tests (hence, 2 times the same) - for `expect.js`'s `takeScreenshot` and that minimodule.

I don't feel like expanding the test logic by extra 3-4 lines is going to hurt its legibility:

```js
ezFileIo = require('../utils/easy-file-io');
ezFileIo.saveRawBase64Data.mockReturnValue('/mock/file/path');
// ...
```

```js
jest.mock('fs-extra').mock('../utils/temporaryPath');
require('../utils/temporaryPath').for.png.mockReturnValue('mock.png');
// ...
expect(require('fs-extra').writeFile).toHaveBeenCalledWith('mock.png', invokeResultInBase64, 'base64');
```